src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -212,5 +230,22 @@ private IEnumerable<IConfigurationSection> GetFeatureDefinitionSections()
                 return _configuration.GetChildren();
             }
         }
+
+        private T ParseFeatureDefinitionSectionEnum<T>(IConfigurationSection configurationSection, string keyword, T enumValue)
+            where T : struct, Enum
+        {
+            string rawValue = configurationSection[keyword];
+
+            //
+            // If the enum is specified, parse it and set the return value
+            if (!string.IsNullOrEmpty(rawValue) && !Enum.TryParse(rawValue, ignoreCase: true, out enumValue))
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.InvalidConfigurationSetting,
```

Comment:
The knowledge that configurationSection.Key is feature name is a bit of odd encapsulation. I'd suggest updating the method signature.

```
private T ParseEnum<T>(
    string feature,
    string value,
    T enumValue)
```